src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -33,6 +33,7 @@
 #include <shogun/lib/SGVector.h>
 #include <shogun/mathematics/eigen3.h>
 #include <shogun/mathematics/linalg/LinalgBackendBase.h>
+#include <Eigen/Core>
```

Comment:
Not sure this is the clean way to do this. Maybe rather in eigen3.h
@vigsterkr ?